### PR TITLE
Add a test for decompositions that decomposes all the operations as much as possible

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #92243
* #92004
* __->__ #87182
* #87183
* #92003

This will enable a more thorough testing of the decompositions than the
one just provided by OpInfos.

cc @ezyang @mruberry @ngimel @Lezcano @fdrocha @peterbell10

### CODE REVIEW

**Comment by lezcano** [comment]
Because it takes quite long. Now, it found SO MANY bugs (all this stack is bug fixes for these bugs), that one could make the case that this is actually the way we want to test things. I'll leave it up to you.

File: test/test_decomp.py [hunk 1]
```diff
@@ -412,6 +427,8 @@ def _torch_dispatch(self, func, types, args=(), kwargs=None):
 
                 # Stuff we shouldn't bother testing
                 # (TODO: remove detach from the decomp table?)
+                # N.b. Testing in-place ops would need dedicated logic
+                in_place = func.name()[-1] == '_'
```
Comment (ezyang) [inline test/test_decomp.py:431]: @anjali411 don't we have a more direct way of querying if an OpOverload is inplace now?

File: test/test_decomp.py [hunk 2]
```diff
@@ -422,7 +439,7 @@ def _torch_dispatch(self, func, types, args=(), kwargs=None):
                     torch.ops.aten.new_empty_strided.default,
                     torch.ops.aten.randn.default,
                     torch.ops.aten.native_dropout.default,
-                ] or any_unsupported(args, kwargs):
+                ] or any_unsupported(args, kwargs) or in_place:
```
Comment (ezyang) [inline test/test_decomp.py:442]: doesn't this change the behavior of non-recursive decomps too?

**Review by ezyang — APPROVED** [review]

**Comment by lezcano** [comment]
Actually, I'm tempted to make this the default behaviour. It's slow, but it's not slower than most tests in opinfos really (think gradtest). Also, the wins from finding these issues early are massive. What do you think @ezyang?

**Comment by ezyang** [comment]
I am for it

**Comment by ezyang** [comment]
Get rid of the old test whole youre at it

**Comment by Chillee** [comment]
I'm also for it. I actually thought this was how we usually tested :P

File: test/test_decomp.py (hunk 2 above)
Comment (lezcano) [reply test/test_decomp.py:442]: Niet. in-place ops were not being tested, before, and those that were, happened to pass this by chance. There is no logic in this function to handle in-place functions.

**Pull request closed without merging**